aries_cloudagent/resolver/base.py

```diff
@@ -95,8 +120,23 @@ async def resolve(self, profile: Profile, did: Union[str, DID]) -> DIDDocument:
                 vm_allow_type_list,
             },
         )
-        return result
+        return Resolution(result, resolver_metadata)
 
     @abstractmethod
     async def _resolve(self, profile: Profile, did: str) -> dict:
         """Resolve a DID using this resolver."""
+
+    def _retrieve_resolver_metadata(self, method, previous_time):
+
+        time_now = datetime.utcnow()
+        time_now.strftime("%Y-%m-%dT%H:%M:%SZ")
+        duration = int((time_now - previous_time).total_seconds() * 1000)
+        resolver_metadata = {
+            "type": self.type,
+            "driverId": f"did:{method}",
+            "driver": self.driver,
+            "retrieved": time_now,
+            "duration": duration,
+        }
+
+        return resolver_metadata
```

Comment:
If this is intended for plugins to implement, should it be `async` or `@abstractmethod`?

Reply:
In my view is just a metadata retriever. Every plugin will retrieve different data but in the same data structure.  where we can find the advantage of doing it async?

Reply:
since `self.type_` and `self.driver` is guaranteed, I think your right. We gain nothing from `async` and there is no need for `@abstractmethod`. What you have looks good.